src/main/java/org/apache/maven/resolver/internal/ant/tasks/CreatePom.java

```diff
@@ -0,0 +1,346 @@
+     *
+     * @param name the name for the POM
+     */
+    public void setName(String name) {
+        this.name = getProject().replaceProperties(name);
+    }
+
+    /**
+     * Get the description for the POM.
+     *
+     * @return the description for the POM
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Set the description for the POM.
+     * The description can include properties that will be replaced at runtime.
+     *
+     * @param description the description for the POM
+     */
+    public void setDescription(String description) {
+        this.description = getProject().replaceProperties(description);
+    }
+
+    /**
+     * skipPomRegistration is a flag to indicate whether the POM should be registered in the project after creation.
+     * Setting this to false is equivalent to doing
+     * <pre><code>
+     *   &lt;pom file="${pomFile}"/&gt;
+     * </code></pre>
+     * Default is false, meaning the POM will be registered.
+     *
+     * @param skipPomRegistration true to skip registration, false to register the POM
+     */
+    public void setSkipPomRegistration(boolean skipPomRegistration) {
+        this.skipPomRegistration = skipPomRegistration;
+    }
+
+    /**
+     * Get the groupId.
+     * If the groupId is not set, it will return a default value based on the project property 'groupId'.
+     *
+     * @return the groupId for the POM
+     */
+    public String getGroupId() {
+        if (groupId == null) {
+            return getProject().getProperty("groupId");
+        }
+        return groupId;
+    }
+
+    /**
+     * Get the artifactId.
+     * If the artifactId is not set, it will return a default value based on the project property 'artifactId'.
+     *
+     * @return the artifactId for the POM
+     */
+    public String getArtifactId() {
+        if (artifactId == null) {
+            return getProject().getProperty("artifactId");
+        }
+        return artifactId;
+    }
+
+    /**
+     * Get the version.
+     * If the version is not set, it will return a default value based on the project property 'version'.
+     *
+     * @return the version for the POM
+     */
+    public String getVersion() {
+        if (version == null) {
+            return getProject().getProperty("version");
+        }
+        return version;
+    }
+
+    /**
+     * Get the description.
+     * If the description is not set, it will return a default value based on the project property 'description'.
+     *
+     * @return the description for the POM
+     */
+    public String getDescription() {
+        return description == null ? super.getDescription() : description;
+    }
+
+    /**
+     * Add a licenses element to the POM.
+     * This allows you to specify one or more licenses that the project is distributed under.
+     *
+     * @param licenses the licenses to add to the POM
+     */
+    public void addLicenses(Licenses licenses) {
```

Comment:
setLicenses

Reply:
License is a nested element so must be named addXXX, setLicenses will not work (ant requirement).